### PR TITLE
PEP8 changes and code structure refactor

### LINKED ISSUE #11: Add functionality to provide output path as well instead of assuming it
Output path is currently set as ```converted-text``` inside the source directory. Instead we should be taking this information from the user

### PR DESCRIPTION
**Pull request by suresttexas00**
This changes behavior slightly for the output destination:
 - default output path is the directory from which you ran main.py, not the pictures source directory (plus the default subdirectory `/converted-text/`)
 - added a -o "output" option for passing a different output directory
 - Make argparse support richer `main.py --help`

PEP-8 changes:
- space indentation and other whitespace corrections
- remove cases of shadowing the outer scope in variable names.

### CODE REVIEW

**Comment by prabhakar267** [comment]
@suresttexas00 the fix for #11 was already in PR when you raised yours.
Kindly look into 04cad8e5e52dc7b26f6f250fb3c2e0558419d007 and if you could update your PR just for the PEP8 changes

It was a good suggestion to enforce PEP8, maybe we could add it in `travis.yml` to run PEP8 conformance (maybe using [flake8](http://flake8.pycqa.org/en/latest/))

**Comment by suresttexas00** [comment]
I have updated the PR.  It is more than just PEP-8 (I only did PEP-8 because I touched the code in the first place).
-  The help and arguments rely less on a second positional argument, using a flag instead (-o)
-  The default behavior uses the current working directory versus using the source files directory for output.

**Comment by suresttexas00** [comment]
Also, have you looked into this: [Issue 15 (dependencies)](https://github.com/prabhakar267/ocr-convert-image-to-text/issues/15)?

File: main.py [hunk 1]
```diff
@@ -1,66 +1,89 @@
...
-		print("tesseract-ocr missing, use sudo apt-get install tesseract-ocr to resolve")
-	elif check_path(input_path):
 
-		count = 0
-		other_files = 0
+def main(source, output_dir):
+    current_directory = os.getcwd()
+    destination = output_dir
+    if not check_path(destination):
+        destination = current_directory + output_dir
```
Comment (prabhakar267) [inline main.py:32]: use `os.path.join()`

File: main.py [hunk 2]
```diff
@@ -1,66 +1,89 @@
...
+        create_directory(destination)
+        if not check_path(destination):
+            print(
+                "I could not create or find '%s'" % destination
```
Comment (prabhakar267) [inline main.py:36]: Do not use message like this, use something like "Directory could not be created at %s"

File: main.py [hunk 3]
```diff
@@ -1,66 +1,89 @@
...
-				call(["tesseract", image_file_name, text_file_path], stdout=FNULL)
+            if ext.lower() not in VALID_IMAGES:
+                other_files += 1
+                continue
+            else:
+                count += 1
+                image_file_name = source + '/' + f
+                filename = os.path.splitext(f)[0]
+                filename = ''.join(e for e in filename if e.isalnum() or e == '-')
+                text_file_path = "%s/%s" % (destination, filename)
```
Comment (prabhakar267) [inline main.py:59]: same, use `os.path.join`

File: main.py [hunk 4]
```diff
@@ -1,66 +1,89 @@
...
-		output_path = os.path.abspath(args.output_dir)
-	else:
-		output_path = os.path.join(input_path,'converted-text')
-	main(input_path, output_path)
\ No newline at end of file
+    parser = argparse.ArgumentParser(
+        description='Ocr-convert-image-to-text arguments'
+    )
+    parser.add_argument(
+        'input_dir', type=str, default='.', help='Path to directory of images'
```
Comment (prabhakar267) [inline main.py:78]: add `required` parameter instead of giving a `default`

File: main.py [hunk 5]
```diff
@@ -1,66 +1,89 @@
...
+    )
+    parser.add_argument(
+        '--output', "-o", type=str, default="/converted-text",
+        help='Output path for text files'
+    )
+
+    args = parser.parse_args()
+    src_path = os.path.abspath(args.input_dir)
```
Comment (prabhakar267) [inline main.py:86]: have consistent variable names, either use `src_path` or `input_dir`

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

File: main.py (hunk 2 above)
Comment (suresttexas00) [reply main.py]: Why?

File: main.py (hunk 1 above)
Comment (suresttexas00) [reply main.py]: Why?

**Comment by suresttexas00** [comment]
It is obvious that you have very strong "preferences".  Feel free to make the changes you have suggested.

Thanks for sharing this interesting code.  I recall back in the 1990's my company paid over $200 USD for a license for an OCR reader program... :)

**Pull request closed without merging**